bin/update_cucumber

```diff
@@ -109,27 +117,44 @@ class {
     }
 
     /**
-     * @return list<GitHubReleaseArray>
+     * @phpstan-return non-empty-list<TGitHubReleaseArray>
```

Comment:
I noticed that later on we seem to assume the there must be at least one revision (the line with `return array_pop($releases);` would return `null` instead of `array|false`).

Of course if it is expected to have 0 releases, then maybe instead of the assertion and non-empty-list, the return line should be `return array_pop($releases) ?? false;` (or throw an exception). @acoulton what are your thoughts?

Reply:
Well we know that there will be at least one revision, because we are querying github for a specific package that already has releases. So we are never going to get a list shorter than the one we currently get.

As per the inline comment, the array_pop is a workaround for the fact that the `24.1.0` release is not listed on the new repo, so the script would not be able to find the original "current version" in the list of releases to know that `25.0.0` is the next one. It should blow up (rather than returning false) if somehow it got an empty array. Once #306 is merged that workaround can be deleted anyway as it will be irrelevant.

Reply:
OK, in that case I'll keep the code as is.